src/utils/utils.py

```diff
@@ -187,108 +190,75 @@ def load_lines(filename: str) -> Iterable[str]:
-            if t == '':
-                continue
-            if t in tag_to_ix_dict:
-                indexes.append(tag_to_ix_dict[t])
-            else:
-                # index 0 will be used for missing value
-                highest_ix = len(tag_to_ix_dict)
-                new_index = highest_ix + 1
-                tag_to_ix_dict[t] = new_index
-                ix_to_tag_dic[new_index] = t
-                indexes.append(new_index)
-        return indexes
-
-    with codecs.open(data_file, 'r', 'utf-8', errors='ignore') as data_fh:
-        for _ in range(skip_rows):
-            data_fh.readline()
-        for row_idx, row in enumerate(data_fh):
-            try:
-                row = row.rstrip().split(delimiter)
-                sent1 = process_sentence(row[s1_idx], max_seq_len)
-                if targ_map is not None:
-                    targ = targ_map[row[targ_idx]]
-                elif targ_fn is not None:
-                    targ = targ_fn(row[targ_idx])
-                else:
-                    targ = int(row[targ_idx])
-                sent2 = process_sentence(row[s2_idx], max_seq_len)
-                sent2s.append(sent2)
-
-                sent1s.append(sent1)
-                targs.append(targ)
-
-                lex_sem_sample = tags_to_ixs(row[0], lex_sem_to_ix_dic, ix_to_lex_sem_dic)
-                pr_ar_str_sample = tags_to_ixs(row[1], pr_ar_str_to_ix_dic, ix_to_pr_ar_str_dic)
-                logic_sample = tags_to_ixs(row[2], logic_to_ix_dic, ix_to_logic_dic)
-                knowledge_sample = tags_to_ixs(row[3], knowledge_to_ix_dic, ix_to_knowledge_dic)
-
-                idxs.append(row_idx)
-                lex_sem.append(lex_sem_sample)
-                pr_ar_str.append(pr_ar_str_sample)
-                logic.append(logic_sample)
-                knowledge.append(knowledge_sample)
-
-            except Exception as e:
-                print(e, " file: %s, row: %d" % (data_file, row_idx))
-                continue
-
-    ix_to_lex_sem_dic[0] = "missing"
-    ix_to_pr_ar_str_dic[0] = "missing"
-    ix_to_logic_dic[0] = "missing"
-    ix_to_knowledge_dic[0] = "missing"
-
-    lex_sem_to_ix_dic["missing"] = 0
-    pr_ar_str_to_ix_dic["missing"] = 0
-    logic_to_ix_dic["missing"] = 0
-    knowledge_to_ix_dic["missing"] = 0
-
-    return {'sents1': sent1s,
-            'sents2': sent2s,
-            'targs': targs,
-            'idxs': idxs,
-            'lex_sem': lex_sem,
-            'pr_ar_str': pr_ar_str,
-            'logic': logic,
-            'knowledge': knowledge,
+    try:
+        rows = pd.read_csv(data_file, \
+                            sep=delimiter, \
+                            error_bad_lines=False, \
+                            header=None, \
+                            skiprows=skip_rows, \
+                            quoting=csv.QUOTE_NONE,\
+                            encoding='utf-8')
+        rows = rows.fillna('')
+        def targs_to_idx(index):
```

Comment:
It seems weird to re-implement this functionality - can we just use an AllenNLP `Vocabulary` object here as we do elsewhere in the code?